lib/backend_common/backend_common/notifications.py

```diff
@@ -17,3 +26,69 @@
 URGENCY_LEVELS = [
     'LOW', 'NORMAL', 'HIGH',
 ]
+
+
+def get_current_app_credentials() -> dict:
+    return {
+        'id': current_app.config.get('TASKCLUSTER_CLIENT_ID'),
+        'key': current_app.config.get('TASKCLUSTER_ACCESS_TOKEN'),
+        'algorithm': 'sha256',
+    }
+
+
+def generate_random_uid() -> str:
+    return ''.join(random.choice(string.ascii_letters + string.digits) for _ in range(32))
+
+
+def verify_policy_structure(policy: dict) -> None:
+    if any(key not in policy for key in ['frequency', 'identity', 'start_timestamp', 'stop_timestamp', 'urgency']):
+        raise KeyError('Policy missing required key')
+
+    if any(freq_key not in policy['frequency'] for freq_key in ['days', 'hours', 'minutes']):
+        raise KeyError('Policy frequency missing required key')
+
+
+def create_new_nagbot_message(message: str, short_message: str, deadline: datetime, policies: List[dict], uid: str=None) -> str:
+    '''
+    Instantiates a new message to be sent repeatedly by NagBot
+
+    :param message: Long description of message (ie email body)
+    :param short_message: Short description of message (ie email subject, IRC message)
+    :param deadline: Message expiry date
+    :param policies: Notification policies described in dict format
+    :param uid: Optionally specify tracking uid. A random uid will be generated if not given
+
+    :return: Tracking uid for the notification
+    '''
+    for policy in policies:
+        verify_policy_structure(policy)
+
+    if uid is None:
+        uid = generate_random_uid()
+
+    request_url = current_app.config.get('RELENG_NOTIFICATION_POLICY_ENDPOINT') + '/message/' + uid
```

Comment:
I would suggest we change this line to 

```
request_url = current_app.config['RELENG_NOTIFICATION_POLICY_ENDPOINT'] + '/message/' + uid
```

since we don't want to silently fail because of the wrong config.

Also, can you change ``RELENG_NOTIFICATION_POLICY_ENDPOINT`` to ``RELENG_NOTIFICATION_POLICY_URL`` since we use similar naming in other services. 

I would also make ``RELENG_NOTIFICATION_POLICY_URL`` a required variable in ``settings.py``.